.github/workflows/run-validation-on-main.yml

```diff
@@ -0,0 +1,113 @@
+name: Run validation on main branch
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  TEST_FULL_ACR_NAME: ${{ vars.TEST_ACR_NAME }}.azurecr.io
+  TEST_IMAGE_REPOSITORY: github-actions/container-app
+
+jobs:
+  use-builder:
+    name: 'Validating builder scenario'
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    env:
+      TEST_IMAGE_TAG: 'bs-${{ github.run_id }}'
+      TEST_CONTAINER_APP_NAME: 'gh-ca-bs-${{ github.run_id }}'
+
+    steps:
+      - name: Clone Oryx repository
+        uses: actions/checkout@v3
+        with:
+          repository: microsoft/Oryx
+
+      - name: Log in to Azure
+        uses: azure/login@v1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Execute Azure Container Apps Build and Deploy Action
+        uses: Azure/container-apps-deploy-action@main
+        with:
+          appSourcePath: '${{ github.workspace }}/tests/SampleApps/DotNetCore/NetCore6PreviewWebApp'
+          acrName: ${{ vars.TEST_ACR_NAME }}
+          containerAppName: ${{ env.TEST_CONTAINER_APP_NAME }}
+          resourceGroup: ${{ vars.TEST_RESOURCE_GROUP_NAME }}
+          imageToBuild: ${{ env.TEST_FULL_ACR_NAME }}/${{ env.TEST_IMAGE_REPOSITORY }}:${{ env.TEST_IMAGE_TAG }}
+          disableTelemetry: ${{ vars.TEST_DISABLE_TELEMETRY }}
+
+      - name: Delete created Azure Container App

```

Comment:
I don't have a good answer for that, but if any test fail, I assume we will need some manual cleanup, right?
We could have a pre-step that tries to delete any remaining resources maybe? I don't know if there are any better solutions.

Reply:
Updated the workflow so the cleanup steps always run.